src/script/builder/tir/prim_func_frame.h

```diff
@@ -55,7 +59,25 @@ class PrimFuncFrame : public TIRFrame {
 
 PrimFuncFrame PrimFunc_(String name);
 tvm::tir::Var Arg(String name, tvm::tir::Var var);
-tvm::tir::Buffer Arg(String name, tvm::tir::Buffer buffer);
+tvm::tir::Var Arg(String name, tvm::tir::Buffer buffer);
+DictAttrs FuncAttrs(DictAttrs attrs);
+tvm::Type FuncRet(tvm::Type ret_type);
+
+tvm::tir::Buffer MatchBuffer(ObjectRef param, Array<PrimExpr> shape,
+                             DataType dtype = DataType::Float(32),
+                             tvm::tir::Var data = NullValue<tvm::tir::Var>(),
```

Comment:
We use `NullValue` only in legacy cases. Modern TVM use `Optional<>` type instead :-) 

```suggestion
                             Optional<tvm::tir::Var> data = NullOpt,
```